.github/workflows/test.yml

```diff
@@ -4,29 +4,103 @@ on: [push, pull_request]
 
 env:
   CARGO_TERM_COLOR: always
+  # CI builds are closer to a from-scratch build anyways
+  # and incremental compilation hurts reproducibility
+  CARGO_INCREMENTAL: 0
```

Comment:
I'd remove the comment; the commit message gives the reason.

Also please sort CARGO_INCREMENTAL before CARGO_TERM_COLOR